pep-0677.rst

```diff
@@ -0,0 +1,854 @@
+  the additional features, they should be straightforward to propose
+  in the future.
+- We realized that because of overloads, it is not possible to replace
+  all need for Callback Protocols even with an extended syntax. This
+  makes us prefer proposing a simple solution that handles most use
+  cases well.
+
+We confirmed that the current proposal is forward-compatible with
+extended syntax by
+`implementing <https://github.com/stroxler/cpython/tree/callable-type-syntax--extended>`_
+a quick-and-dirty grammar and AST on top of the grammar and AST for
+the current proposal.
+
+
+Syntax Closer to Function Signatures
+------------------------------------
+
+One alternative we had floated was a syntax much more similar to
+function signatures.
+
+In this proposal, the following types would have been equivalent::
+
+    class Function(typing.Protocol):
+        def f(self, x: int, /, y: float, *, z: bool = ..., **kwargs: str) -> bool:
+            ...
+
+    Function = (x: int, /, y: float, *, z: bool = ..., **kwargs: str) -> bool
+
+
+The benefits of this proposal would have included:
+
+- Perfect syntactic consistency between signatures and callable types.
+- Support for more features of function signatures (named, optional,
+  variadic args) that this PEP does not support.
+
+Key downsides that led us to reject the idea include the following:
+
+- A large majority of use cases only use positional-only arguments,
+  and this syntax would be more verbose for that use case, both
+  because of requiring argument names and an explicit ``/``, for
+  example ``(int, /) -> bool`` where our proposal allows ``(int) ->
+  bool``
+- The requirement for explicit ``/`` for positional-only arguments has
+  a high risk of causing frequent bugs - which often would not be
+  detected by unit tests - where library authors would accidentally
+  use types with named arguments.
+- Our analysis suggests that support for ``ParamSpec`` is key, but the
+  scoping rules laid out in PEP 612 would have made this difficult.
+
+
+Other Proposals Considered
+--------------------------
+
+Functions-as-Types
+~~~~~~~~~~~~~~~~~~
+
+An idea we looked at very early on was to `allow using functions as
+types
+<https://docs.google.com/document/d/1rv6CCDnmLIeDrYlXe-QcyT0xNPSYAuO1EBYjU3imU5s/edit?usp=sharing>`_.
+The idea is allowing a function to stand in for its own call
+signature, with roughly the same semantics as the ``__call__`` method
+of Callback Protocols. Think this may be a great idea and worth its
+own PEP, but that it is not a good alternative to improving the
+usability of callable types:
+
+- Using functions as types would not give us a new way of describing
+  function types as first class values. Instead, they would require a
+  function definition statement that effectively defines a type alias
+  (much as a Callable Protocol class statement does).
+- Functions-as-types would support almost exactly the same features
```

Comment:
This right now reads positively. I was left a bit confused since this looked like a list of "Cons".

Maybe rephrase as "They would not support any more features than callback protocols while only being somewhat shorter."